riklet/src/core.rs

```diff
@@ -166,9 +172,24 @@ impl Riklet {
             if !Path::new(&file_path).exists() {
```

Comment:
You could work with `PathBuf` from the start, it would make the code a bit easier to read 

```suggestion
           let file_pathbuf = Path::new(&file_path)
            if !file_pathbuf.exists() {
```